src/main/core/manager.c

```diff
@@ -409,9 +442,40 @@ static gchar** _manager_generateEnvv(Manager* manager, InterposeMethod interpose
             if (key != NULL && value != NULL) {
                 /* check if the key is LD_PRELOAD */
                 if (!g_ascii_strncasecmp(key, "LD_PRELOAD", 10)) {
-                    /* append all LD_PRELOAD entries */
-                    debug("adding key path %s", value);
-                    g_ptr_array_add(ldPreloadArray, g_strdup(value));
+                    // Handle the list of entries, which could be separated by ' ' or ':'.
+                    GString* preloadBuf = g_string_new(value);
+
+                    // First replace all occurences of ' ' by ':'.
+                    // g_string_replace does what we want, but isn't available until glib 2.68.
+                    for (int i = 0; i < preloadBuf->len; i++) {
+                        if (preloadBuf->str[i] == ' ') {
+                            preloadBuf->str[i] = ':';
+                        }
+                    }
+
+                    // Now split by ':' to handle each preload path.
+                    gchar** paths = g_strsplit(preloadBuf->str, ":", 0);
+
+                    // Expand any paths that begin with '~'.
+                    for (int i = 0; paths != NULL && paths[i] != NULL; i++) {
+                        GString* pathBuf = g_string_new(NULL);
+
+                        if (!g_ascii_strncasecmp(paths[i], "~/", 2)) {
+                            g_string_printf(pathBuf, "%s%s", g_get_home_dir(), &paths[i][1]);
+                        } else if (!g_ascii_strncasecmp(paths[i], "~", 1)) {
+                            g_string_printf(pathBuf, "%s/%s", g_get_home_dir(), &paths[i][1]);
```

Comment:
What's the purpose of this condition? This seems like it doesn't work for paths like `~steve/.local/bin`, which will turn into `/home/steve/steve/.local/bin`.

Reply:
I was trying to avoid expanding to `/home/steve//.local` (with two slashes), but I didn't realize that `~name` is usually expanded to `/home/name`.

I could just remove that condition and then we only support expanding `~/` but not `~`? Or keep the condition as is but change line 466 to
```
g_string_printf(pathBuf, "/home/%s", &paths[i][1]);
```

Which do you prefer?

Reply:
I went withy the second approach :)

Reply:
I'm not really sure what you mean here, or why there would be two slashes, but using `"/home/%s"` looks good to me.